### PR TITLE
SYCL improvements

### PR DESCRIPTION
**Pull request by lukeiwanski**
* Added Tile, Transpose and Range Ops double support for SYCL device.
* Moved gpu_device_name() to test_util.py so now it can be used in force_gpu to pull either GPU or SYCL depending on what is avaiable in the system.

### CODE REVIEW

File: tensorflow/python/framework/test_util.py [hunk 1]
```diff
@@ -43,7 +43,14 @@
 from tensorflow.python.platform import tf_logging as logging
 from tensorflow.python.util import compat
 from tensorflow.python.util.protobuf import compare
+from tensorflow.python.client import device_lib as _device_lib
```
Comment (benoitsteiner) [inline tensorflow/python/framework/test_util.py]: Why not just from tensorflow.python.client import device_lib ?

File: tensorflow/python/platform/test.py [hunk 2]
```diff
@@ -137,15 +137,6 @@ def is_gpu_available(cuda_only=False):
     return any((x.device_type == 'GPU' or x.device_type == 'SYCL')
                for x in _device_lib.list_local_devices())
 
-
-def gpu_device_name():
```
Comment (benoitsteiner) [inline tensorflow/python/platform/test.py]: Aren't the tests that call test.gpu_device_name() going to fail ?

File: tensorflow/python/framework/test_util.py (hunk 1 above)
Comment (lukeiwanski) [inline tensorflow/python/framework/test_util.py]: Fixed in 63aad7c1091268a549a4782a7fa269ff5ce19bb0

File: tensorflow/python/platform/test.py (hunk 2 above)
Comment (lukeiwanski) [inline tensorflow/python/platform/test.py]: Overlooked one change set in the commit. Fixed in 38cfa246e344d20cfc822241ddc5863c4d22d58e

**Pull request merged**